targets/frontend/src/pages/storage/[container]/index.js

```diff
@@ -0,0 +1,327 @@
+          <Select
+            id="sort"
+            name="sort"
+            value={sort}
+            onChange={(e) => setSort(e.target.value)}
+          >
+            <option value="mostRecent">Plus récent</option>
+            <option value="oldest">Plus vieux</option>
+            <option value="alphabetic">Alphabétique</option>
+            <option value="reverse-alphabetic">Alphabétique inversé</option>
+            <option value="heaviest">Plus lourd</option>
+            <option value="lightest">Plus léger</option>
+          </Select>
+        </Flex>
+        <Flex sx={{ flexDirection: "column", ml: "small" }}>
+          <Label htmlFor="filter">Filtrer par type</Label>
+          <Select
+            id="filter"
+            name="filter"
+            label="Filtrer par type"
+            value={filter}
+            onChange={(e) => setFilter(e.target.value)}
+          >
+            <option value="">Tous</option>
+            <option value="jpg">jpg</option>
+            <option value="png">png</option>
+            <option value="pdf">pdf</option>
+            <option value="word">document word</option>
+          </Select>
+        </Flex>
+      </form>
+      {isSearching || !data ? (
+        <Spinner />
+      ) : (
+        <ul sx={{ listStyleType: "none", m: 0, p: 0 }}>
+          {displayedFiles.map((file) => {
+            return (
+              <li key={file.name}>
+                <Card
+                  as="a"
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  href={file.url}
+                  key={file.name}
+                  title={file.name}
+                  sx={{
+                    alignItems: "center",
+                    color: "text",
+                    display: "flex",
+                    justifyContent: "space-between",
+                    mt: "small",
+                    textDecoration: "none",
+                  }}
+                >
+                  <IoMdDownload
+                    sx={{ flex: "0 0 auto", mr: "small", ...iconSx }}
+                  />
+                  <div sx={{ flex: "1 1 auto", minWidth: 0 }}>
+                    <div
+                      sx={{
+                        fontSize: "1.1rem",
+                        fontWeight: "bold",
+                        overflow: "hidden",
+                        textOverflow: "ellipsis",
+                        whiteSpace: "nowrap",
+                      }}
+                    >
+                      {file.name}
+                    </div>
+                    <div sx={{ fontSize: "small" }}>
+                      <span sx={{ fontWeight: "bold" }}>Poids&nbsp;:</span>{" "}
+                      {prettyBytes(file.contentLength)} | Mise en ligne il y a{" "}
+                      {timeSince(file.lastModified)}
+                    </div>
+                  </div>
+                  {hasClipboardApi && (
+                    <Button
```

Comment:
I think a dedicated ClipBoardButton which handle all the dedicated code (feature detection + localState for clipboard ui status update) would be clearer